### DISCUSSION TITLE
tdNavigationDrawerMenu no worker

### DISCUSSION THREAD

**Discussion by ibmoreno** [Teradata/covalent#2085] [General]
The <td-navigation-drawer> tag with <div tdNavigationDrawerMenu> is not allowing <mat-nav-list> to be outside the drawer

example of old documentation
https://teradata.github.io/covalent/v2/#/design-patterns/navigation-drawer

**Comment by owilliams320** [comment]
@ibmoreno can you explain what is going on here in better detail. the navigation drawer has not been changed in quite some time so it should be functioning as is

**Reply by ibmoreno** [reply]
Hi @owilliams320,  Sorry for my English, I'm from Brazil. Well, in my understanding, the Drawer is to hide the menu with the tdNavigationDrawerMenu marking and show the other menu without the marking, however it was always showing both menus when opening <td-navigation-drawer> and not showing anything when clicking on the drawer menu.
I downloaded the source and checked the tdNavigationDrawerMenu directive, I made the following change.

`<td-layout>

    <td-navigation-drawer 
        avatar="assets/icons/icon_d.svg" 
        flex 
        sidenavTitle="sample." 
        name="User System"
        email="user@domain.com.br" 
        navigationRoute="/home">

        <div td-navigation-drawer-menu>
            <mat-nav-list>
                <mat-list-item tdLayoutClose>
                    <mat-icon matListItemIcon>account_box</mat-icon>Gerenciar Conta
                </mat-list-item>
            </mat-nav-list>
        </div>
        <mat-nav-list tdNavigationDrawerMenu>
            <mat-list-item routerLink="/home" routerLinkActive="list-item-active" tdLayoutClose>
                <mat-icon matListItemIcon>home</mat-icon>Home
            </mat-list-item>
            <mat-list-item routerLink="/customers" routerLinkActive="list-item-active" tdLayoutClose>
                <mat-icon matListItemIcon>groups</mat-icon>Clientes
            </mat-list-item>
            <mat-list-item tdLayoutClose>
                <mat-icon matListItemIcon>logout</mat-icon>Sair
            </mat-list-item>
        </mat-nav-list>
    </td-navigation-drawer>
 
    <div td-sidenav-content class="td-layout-footer pad">
        <span class="mat-caption">versão 1.0.0</span>
    </div>

    <td-layout-nav toolbarTitle="controle interno />" navigationRoute="/">

        <button mat-icon-button td-menu-button tdLayoutToggle>
            <mat-icon>menu</mat-icon>
        </button>

        <div td-toolbar-content layout="row" layout-align="end center" flex>
            <button class="overflow-visible" mat-icon-button
                matTooltip="Sair do Sample">
                <mat-icon>logout</mat-icon>
            </button>
        </div>

        <router-outlet></router-outlet>

        <td-layout-footer>
            <div class="mat-caption tc-grey-500" layout="row" layout-align="center">
                <span class="mat-caption">Copyright &copy; 2024. Todos os direitos reservados.</span>
            </div>
        </td-layout-footer>

    </td-layout-nav>

</td-layout>`

As you can see, the tags <td-navigation-drawer-menu> and
<tdNavigationDrawerMenu> to change menus when clicking the drawer. now it's working. I don't know if this is how it has to be, but in the old examples both menus work, in the current example it doesn't.
Another issue now is getting the routerLinkActive to work, which is not standing out when it is on the route, but I will investigate later, if you know anything please, I would be very grateful.

I have been using the teradata layout structure for some time and following its component evolution. They are to be congratulated. a clean and very intuitive layout.



![Captura de tela 2024-01-04 132558](https://github.com/Teradata/covalent/assets/135080/ce4ab443-fec5-42c7-a125-078fdc3e636f)
--
![Captura de tela 2024-01-04 132644](https://github.com/Teradata/covalent/assets/135080/c7c8fb12-f5ea-4fa6-a4b6-45fd5ebdaca4)
